### PR TITLE
add global heap hint setter function

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

**Review by billphipps — DISMISSED** [review]

File: examples/client/client.c [hunk 1]
```diff
@@ -3042,6 +3042,10 @@ THREAD_RETURN WOLFSSL_THREAD client_test(void* args)
             != 0) {
         err_sys("unable to load static memory");
     }
+    /* for test case (does not handle all memory used on default build)
```
[the remark this answers is not in the document]
Comment (JacobBarthelmeh) [reply examples/client/client.c]: The comment is about heap memory size but it also has the issue of some of our tests being threaded and using the global heap hint in other tests. Removed the reference here and we now just have the test in wolfcrypt/test/test.c.

File: wolfcrypt/src/memory.c [hunk 2]
```diff
@@ -910,7 +921,12 @@ void* wolfSSL_Malloc(size_t size, void* heap, int type)
     }
     else {
         WOLFSSL_HEAP_HINT* hint = (WOLFSSL_HEAP_HINT*)heap;
-        WOLFSSL_HEAP*      mem  = hint->memory;
+        WOLFSSL_HEAP*      mem;
+
+        if (hint == NULL) {
+            hint = (WOLFSSL_HEAP_HINT*)globalHeapHint;
```
[the remark this answers is not in the document]
Comment (JacobBarthelmeh) [reply wolfcrypt/src/memory.c:941]: Added in a debug print out

File: wolfssl/wolfcrypt/memory.h [hunk 3]
```diff
@@ -216,6 +216,7 @@ WOLFSSL_API int wolfSSL_GetAllocators(wolfSSL_Malloc_cb* mf,
         byte        haFlag; /* flag used for checking handshake count */
     } WOLFSSL_HEAP_HINT;
 
+    WOLFSSL_API void wolfSSL_SetGlobalHeapHint(void* heap);
```
[the remark this answers is not in the document]
Comment (JacobBarthelmeh) [reply wolfssl/wolfcrypt/memory.h]: Updated

**Review by billphipps — APPROVED** [review]

File: wolfcrypt/src/memory.c [hunk 4]
```diff
@@ -899,6 +899,31 @@ int wolfSSL_GetMemStats(WOLFSSL_HEAP* heap, WOLFSSL_MEM_STATS* stats)
...
+{
+    void *oldHint = globalHeapHint;
+
+    globalHeapHint = heap;
+    return oldHint;
+}
+
+
+/* returns a pointer to the current global heap hint */
+void* wolfSSL_GetGlobalHeapHint()
```
Comment (dgarske) [inline wolfcrypt/src/memory.c:921]: `(void)`.

**Review by dgarske — CHANGES_REQUESTED** [review]
This PR is great. What's holding it up?

**Comment by JacobBarthelmeh** [comment]
Restest this please Jenkins

**Comment by JacobBarthelmeh** [comment]
I don't like that a retest needed done. Will need to keep an eye out for if the tests are having a threading issue with the setting of the global heap hint (is why I limited the set global heap hint test).

**Review by dgarske — APPROVED** [review]
@JacobBarthelmeh please make sure the static memory / user manual adds a few lines about these new APIs. Very handy addition.

**Pull request merged**